doc/developers/cython.rst

```diff
@@ -0,0 +1,178 @@
+.. _cython:
+
+Cython Best Practices, Conventions and Knowledge
+================================================
+
+This documents:
+
+* technical aspects of Cython which aren't currently present in Cython's documentation
+* tips to develop Cython code in scikit-learn
+
+Cython Internals
+----------------
```

Comment:
This section presents advanced stuff. I think it should come after the tips which will be useful in more situations